pkg/mimir/grpc_push_check.go

```diff
@@ -34,7 +34,9 @@ func (g *grpcInflightMethodLimiter) RPCCallStarting(methodName string) error {
 			return errNoIngester
 		}
 
-		return ing.StartPushRequest()
+		if err := ing.StartPushRequest(); err != nil {
+			return status.Error(codes.Unavailable, err.Error())
+		}
```

Comment:
Can you explain what changed here?

Reply:
Sure, the idea of this PR is that no ingester method except `ingester.Push()` returns an HTTP or a gRPC error. Previously all the errors returned by `ingester.StartPushRequest()` were annotated with `codes.Unavailable`. This PR has changed that behavior, and therefore the callers of `ingester.StartPushRequest()` should add an appropriate gRPC code.